community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/steps/triadicSelectionFinder.scala

```diff
@@ -27,43 +27,83 @@ import org.neo4j.cypher.internal.frontend.v2_3.ast._
 import org.neo4j.cypher.internal.frontend.v2_3.ast.Expression
 
 object triadicSelectionFinder extends CandidateGenerator[LogicalPlan] {
-  override def apply(in: LogicalPlan, qg: QueryGraph)(implicit context: LogicalPlanningContext): Seq[LogicalPlan] = in match {
+
+  override def apply(in: LogicalPlan, qg: QueryGraph)(implicit context: LogicalPlanningContext): Seq[LogicalPlan] =
+    qg.selections.patternPredicatesGiven(in.availableSymbols).collect {
```

Comment:
We should make sure to not try to solve predicates that have already been solved, by filtering out predicates already  in `in.solved.lastQueryGraph.selections`